### DISCUSSION TITLE
Being able to group multiple Plugin Configurations together

### DISCUSSION THREAD

**Discussion by subnetmarco** [Kong/kong#7258] [Ideas and feature requests]
Being able to include multiple plugin configurations together under a group, that I can then apply to APIs. In this way I can enforce same rules to different APIs, without duplicating the plugin configurations across the board.

**Comment by Tieske** [comment]
This is what was discussed as part of #505 where the idea was to rename `api` to `route` and have another grouping level `service` which could contain multiple apis. Every `service` would also have a set of plugins.

**Comment by Vermeille** [comment]
Is there anything going forward on this? I'm willing to contribute (depending on the amount of work) since it could benefit my company.

**Comment by thibaultcha** [comment]
Yes, there is work scheduled on this.

**Comment by yuannan-zz** [comment]
> This is what was discussed as part of #505 where the idea was to rename `api` to `route` and have another grouping level `service` which could contain multiple apis. Every `service` would also have a set of plugins.
Hi @Tieske , 
I have two api groups and each group has many apis which route to different hosts and ports . now i want to set jwt plugin for two api groups respectively. Is this the `service` usecase for avoiding duplicating the plugin configurations for every api in group? However, if i group apis in `service`, how can i dispatch the api request according the `path` to proper upstream?

**Comment by Tieske** [comment]
I don't think so, if you have multiple upstream hosts then those would typically each be a service in Kong. You'd still have to duplicate the config on each of those
